core/app/models/concerns/spree/active_storage_adapter/attachment.rb

```diff
@@ -59,12 +61,25 @@ def metadata
         @attachment.metadata
       end
 
-      def normalize_styles(styles)
-        styles.transform_values { |v| v.split('x').map(&:to_i) }
+      def styles_to_transformations(styles)
+        styles.transform_values(&method(:imagemagick_to_image_processing_definition))
+      end
+
+      def imagemagick_to_image_processing_definition(definition)
+        width_height = definition.split('x').map(&:to_i)
+
+        case definition[-1].to_sym
+        when :>
+          { resize_to_limit: width_height }
+        when :^
```

Comment:
This actually has to be

```suggestion
        when :#
```

for `resize_to_fill`, because it crops the image to fill the space. See https://github.com/janko/image_processing/wiki/Resize-To-Fill

Reply:
Actually, `image_processing` transforms `resize_to_fill` into `^`:

https://github.com/janko/image_processing/blob/12e7cf50b512d0f4c0735138013c5ea6d87fdee5/lib/image_processing/mini_magick.rb#L74

See also [docs in imagemagick](https://legacy.imagemagick.org/Usage/resize/#fill).

Do I get it right?

Reply:
Yes. I see, this seems to be a Dragonfly thing. Really confusing actually, because Dragonfly just forwards the string to `convert -thumbnail` https://markevans.github.io/dragonfly/imagemagick#thumb